components/DogCard.js

```diff
@@ -1,29 +1,115 @@
+import React, { useState, useEffect } from "react";
 import styled from "styled-components";
-import { useState } from "react";
+import Image from "next/image";
+import editButtonCardImage from "../public/images/edit-button-card.png";
+
+export default function DogCard({ dog, onDelete, onEdit, onUpdate }) {
+  const [isEditMode, setIsEditMode] = useState(false);
+  const [editedDog, setEditedDog] = useState(dog);
+
+  useEffect(() => {
+    setEditedDog(dog);
+  }, [dog]);
 
-export default function DogCard({ dog, onDelete }) {
   const handleDeleteClick = () => {
     const confirmation = window.confirm(
       "Are you sure you want to delete this dog?"
     );
     if (confirmation) {
-      onDelete(dog.name);
+      onDelete(dog.id);
     }
   };
 
+  const handleEditClick = () => {
+    setIsEditMode(true);
+  };
+
+  const handleSaveClick = () => {
+    const updatedDog = { ...editedDog };
+    onUpdate(updatedDog);
```

Comment:
```suggestion
    onUpdate(editedDog);
```
You could remove the unnecessary variable assignment and directly passe the `editedDog` object to the `onUpdate` callback.